meson.build

```diff
@@ -0,0 +1,521 @@
+              kmod_dep,
+            ],
+            link_with: [
+               pcilib,
+            ],
+            install: true)
+
+setpci_srcs = common_srcs + [
+  'setpci.c',
+]
+
+executable('setpci',
+           sources: setpci_srcs,
+           link_with: [
+              pcilib,
+           ],
+           install: true)
+
+example_srcs = [
+  'lib/config.h',
+  'lib/header.h',
+  'lib/pci.h',
+  'lib/sysdep.h',
+  'lib/types.h',
+  'example.c',
+]
+
+executable('example',
+           sources: example_srcs,
+           link_with: [
+              pcilib,
+           ],
+           install: true)
+
+install_headers(
+  'lib/config.h',
+  'lib/header.h',
+  'lib/pci.h',
+  'lib/types.h',
+  subdir: 'pci',
+)
+
+custom_target('update-pciids',
+              build_by_default: true,
+              command: [
+                'sed',
+                '-e',
+	        's@^DEST=.*@DEST="' + IDSDIR/pci_ids + '"@',
+                '-e',
+                's@^PCI_COMPRESSED_IDS=.*@PCI_COMPRESSED_IDS='
+                  + zlib_dep.found().to_int().to_string()
+                  + '@',
+                '-e',
+                's@VERSION=.*@VERSION=' + meson.project_version() + '@',
+                '@INPUT@',
+              ],
+              capture: true,
+              input: [
+                'update-pciids.sh',
+              ],
+              output: [
+                'update-pciids',
+              ],
+              install: true,
+              install_dir: get_option('sbindir'),
+              install_mode: ['rwxr-xr-x'])
+
+if pci_ids == 'pci.ids.gz'
+  custom_target(pci_ids,
+                build_by_default: true,
+                command: [
+                  'gzip',
+                  '-9nc',
+                  '@INPUT@'
+                ],
+                capture: true,
+                input: [
+                  'pci.ids',
+                ],
+                output: [
+                  pci_ids
+                ],
+                install: true,
+                install_dir: IDSDIR,
+                install_mode: ['rw-r--r--'])
+endif
+
+build_date = get_option('build-date')
+if build_date == ''
+  build_date = run_command(['date', '+%d-%B-%Y'],
+                           check: false,
+                           capture: true).stdout().strip()
+endif
+
+custom_target('pci.ids.5',
+              build_by_default: true,
+              command: [
+                'sed',
+                '-e',
+                's/@TODAY@/' + build_date + '/',
+                '-e',
+                's/@VERSION@/pciutils-' + meson.project_version() + '/',
+                '-e',
+                's#@IDSDIR@#' + IDSDIR + '#',
+                '-e',
+                's#@PCI_IDS@#' + pci_ids + '#',
+                '@INPUT@'
+              ],
+              capture: true,
+              input: [ 'pci.ids.man' ],
+              output: [ 'pci.ids.5' ],
+              install: true,
+              install_dir: get_option('mandir') / 'man5')
```

Comment:
You can get rid of all these seds using a meson builtin approach to configuring files:

```meson
replacements = {
  'TODAY': build_date,
  'VERSION': 'pciutils-'+meson.project_version(),
  'IDSDIR': IDSDIR,
  'PCI_IDS': pci_ids,
}

foreach man: ['pci.ids.5', 'pcilib.7', 'lspci.8', 'setpci.8', 'update-pciids.8']
    manfile = configure_file(
        input: man.substring(0, -1)+'man',
        output: man,
        configuration: replacements
    )
    install_man(manfile)
endforeach
```